docs/UserGuide.md

```diff
@@ -218,19 +220,50 @@ Examples:
 
 ## Features: Managing Person Relationships
 
+### Listing all relationships : `listRelation`
+
+Shows a list of all current relationships in the address book.
+
+Format: `listRelation`
+
 ### Adding a relationship : `addRelation`
 
-Adds a relationship between two people in the address book.
+Adds a roleless relationship between two people in the address book.
 
 Format: `addRelation UUID1 UUID2 RELATIONSHIP_TYPE`
 
-* Adds the relationship between the person with the specified `UUID1` and the person with the specified `UUID2`.
+* Adds the roleless relationship between the person with the specified `UUID1` and the person with the specified `UUID2`.
 * The `UUID1` and `UUID2` refer to the unique identifiers of the persons shown in the displayed person list.
 * The `UUID1` and `UUID2` **must be valid UUIDs**.
 * The `RELATIONSHIP_TYPE` is case-sensitive and matches the relationship type name exactly.
 * If the relationship already exists, the command will not have any effect.
 * If either persons do not exist, the command will not have any effect.
 
+Examples:
+* `addRelation 12db 34ab friend` adds the relation friend between the person with the UUID 12db and the person with the UUID 34ab.
+
+Adds a role-based relationship between two people in the address book.
+
+Format: `addRelation ROLE1 UUID1 ROLE2 UUID2 RELATIONSHIP_TYPE`
+
+* Adds the role-based relationship between the person with the specified `UUID1` and `ROLE1` and the person with the specified `UUID2` and `ROLE2`.
+* The `UUID1` and `UUID2` refer to the unique identifiers of the persons shown in the displayed person list.
+* The `UUID1` and `UUID2` **must be valid UUIDs**.
+* The `ROLE_1` and `ROLE_2` are case-sensitive and match the role name exactly.
```

Comment:
Roles are also not case-sensitive any more for any of the relationship commands, must be string though.